File: src/main/java/com/github/kory33/signvote/exception/VotePointAlreadyVotedException.java
package com.github.kory33.signvote.exception;

import com.github.kory33.signvote.model.VotePoint;
import lombok.EqualsAndHashCode;
import lombok.Value;

import java.util.UUID;

/**
 * Represents an exception thrown when a player attempts to vote to a vote point
 * which has already been voted by the same player.
 */
@EqualsAndHashCode(callSuper = false)
@Value
public class VotePointAlreadyVotedException extends Exception {
    UUID voterUUID;
    VotePoint votePoint;
}

File: src/main/java/com/github/kory33/signvote/exception/VotePointAlreadyVotedException.kt
package com.github.kory33.signvote.exception

import com.github.kory33.signvote.vote.VotePoint

import java.util.UUID

/**
 * Represents an exception thrown when a player attempts to vote to a vote point
 * which has already been voted by the same player.
 */
data class VotePointAlreadyVotedException
constructor(val voterUUID: UUID, val votePoint: VotePoint) : Exception()
